framework/src/functions/PiecewiseConstant.C

```diff
@@ -67,7 +67,13 @@ PiecewiseConstant::value(Real t, const Point & p)
   const unsigned len = functionSize();
   const Real toler = 1e-14;
 
-  // endpoint case
+  // endpoint cases
+  if ( (_direction == LEFT  && x < (1+toler) * domain(0)) ||
+       (_direction == RIGHT && x < (1-toler) * domain(0)) )
```

Comment:
Yes, whenever the function is queried to the left of the left-most data
point, return the left-most value.

On Mon, Apr 20, 2015 at 11:08 AM, John W. Peterson <notifications@github.com
> wrote:

> In framework/src/functions/PiecewiseConstant.C
> <https://github.com/idaholab/moose/pull/4949#discussion_r28707842>:
>
> > @@ -67,7 +67,13 @@ PiecewiseConstant::value(Real t, const Point & p)
> >    const unsigned len = functionSize();
> >    const Real toler = 1e-14;
> >
> > -  // endpoint case
> > +  // endpoint cases
> > +  if ( (_direction == LEFT  && x < (1+toler) * domain(0)) ||
> > +       (_direction == RIGHT && x < (1-toler) * domain(0)) )
>
> If I understand correctly, this assigns range(0) to x values arbitrarily
> far to the "left" of domain(0) as long as _direction==RIGHT? In other
> words, we want to interpolate values outside the domain to the left
> end-point value, not throw an error?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/4949/files#r28707842>.
>



-- 
Jason D. Hales, PhD
Department Manager
Fuels Modeling & Simulation
Idaho National Laboratory
(208) 526-2293